docs/source/tutorials/extending_mantid_with_python/python_fit_functions/03_attributes.rst

```diff
@@ -45,3 +45,25 @@ If defined in your class then you can use it to set a python attribute on your o
             if name == "NLoops":
                 # Can the be accessed quicker later using self._nloops
                 self._nloops = value
+
+If desired, the possible values that an attribute can take can be restricted through the use
+of a Function Attribute Validator. Such a validator must derive from the ``IValidator``
+abstract class; a number of these are currently provided in the Manitd Kernel such as:
+    - List Validator (attribute value must be specified from a provided list of values).
+    - Bounded Validator (numeric attritute value must be between two provided bounds).
+    - String contrains validator (string attribute value must contain a provided sub-string/s)
+
+In python, attributes can be declared with a validator using the following syntax:
+
+.. code-block:: python
+    from mantid.kernel import StringListValidator, StringContainsValidator, FloatBoundedValidator
+
+    class Example1DFunction(IFunction1D): # or IPeakFunction
+
+        def init(self):
+            self.declareParameter("StrAttr", "acceptable"", StringListValidator(["acceptable","values"]))
+            self.declareAttribute("FloatAtt", 3.0, FloatBoundedValidator(0.0, 5.0))
+            self.declareAttribute("StringContainsAtt", "StringContains", StringContainsValidator(["Contains"]))
+
+If a string list validator is used for any function called fromm the ``FitPropertyBrowser``, the attribute
```

Comment:
Typo in the word 'from'. I also think this line is being picked up as a block of python code - so you might need another empty line above this line

Reply:
Amended typo, don't see the line being picked up as python code on my end or on the rich diff, did you build it?

Reply:
I can see what you mean now, looks like it due to the colour of list/is/for etc. on the git raw text display. Don't think this is an actual issue, but am going to double space the lines as it probably aid readability.